GiniVision/Classes/CameraViewController.swift

```diff
@@ -95,7 +100,12 @@ public typealias CameraErrorBlock = (_ error: CameraError) -> ()
         guard let image = cameraOverlayImage, let cgImage = image.cgImage else {
             return nil
         }
-        return UIImage(cgImage: cgImage , scale: 1.0, orientation: UIDevice.current.orientation.isLandscape ? .right : UIImageOrientation.up)
+        if UIDevice.current.orientation.isFlat {
+            return UIImage(cgImage: cgImage , scale: 1.0, orientation: UIApplication.shared.statusBarOrientation.isLandscape ? .right : UIImageOrientation.up)
```

Comment:
As we discussed before, statusBarOrientation always returns a value even if status bar is hidden. The deviceOrientationMapping is needed because UIInterfaceOrientation has not changed when viewWillTransition is called. And the interfaceOrientationMapping is needed because It could be the case that there is no UIDeviceOrientation (and you need it on initialization).